examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/ntp.c

```diff
@@ -0,0 +1,301 @@
+{
+    WiFi_Status_t status = WiFi_MODULE_SUCCESS;
+    uint8_t sntp_protocol = 'u'; //UDP
+    printf("\r\n\tStablishing UDP connection to SNTP server %s:%lu",
+           sntp_server,
+           sntp_port);
+    status = wifi_socket_client_open((uint8_t*)sntp_server, sntp_port,
+                                     &sntp_protocol, sock_id);
+
+    return status;
+}
+
+/**
+   * @brief
+   * @param  sock_id is the socket ID set by sntp_connect() in stnp_start()
+   * @retval WiFi_Status_t: WiFi_MODULE_SUCCESS on success, see wifi_interface.h
+   */
+WiFi_Status_t sntp_disconnect( uint8_t sock_id )
+{
+    WiFi_Status_t status = WiFi_MODULE_SUCCESS;
+    printf("\r\n>>Closing UDP connection to SNTP server...");
+    status = wifi_socket_client_close(sock_id);
+    if(status != WiFi_MODULE_SUCCESS)
+    {
+        printf("\r\n\tUDP Socket Close [FAIL] Status code: %d", status);
+        return status;
+    }
+    else
+    {
+        printf("\r\n\tUDP Socket Close [OK]");
+    }
+    return status;
+}
+
+/**
+   * @brief  Send SNTP packet to the Server. Must be called after a connection
+   *         has been stablished using sntp_get_datetime()
+   * @param  sock_id is the socket ID set by sntp_connect() in stnp_start()
+   * @retval WiFi_Status_t: WiFi_MODULE_SUCCESS on success, see wifi_interface.h
+   */
+WiFi_Status_t sntp_send_request( uint8_t sock_id )
+{
+    WiFi_Status_t status = WiFi_MODULE_SUCCESS;
+    char sntp_request[SNTP_MSG_SIZE];
+    sntp_build_request(sntp_request);
+
+    printf("\r\n>>Sending SNTP request to the server");
+    status = wifi_socket_client_write(sock_id, SNTP_MSG_SIZE, sntp_request);
+
+    if(status != WiFi_MODULE_SUCCESS)
+    {
+        printf("\r\n\tSNTP Message Write [FAIL] Status code: %d", status);
+    }
+    else
+    {
+        printf("\r\n\tSNTP Message Write [OK]");
+    }
+    return status;
+}
+
+/**
+   * @brief  Convert uint32_t from Network byte order to Host byte order.
+   *         i.e. Return the reverse-endian value of the received argument.
+   *         Implementation lifted from the Cube SDK's LwIP
+   * @param  n is a 32bit integer in network order
+   * @retval Returns a 32bit integer in host order (ready to be used by the uC)
+   */
+uint32_t sntp_ntohl(uint32_t n)
+{
+    return ((n & 0xff) << 24)      |
+           ((n & 0xff00) << 8)     |
+           ((n & 0xff0000UL) >> 8) |
+           ((n & 0xff000000UL) >> 24);
+}
+
+/**
+   * @brief  Parses SNTP response and returns current date and time
+   * @param  48 byte array with the SNTP response. Length MUST be validated
+   *         before calling this function
+   * @retval Current date and time in epoch format, -1 on error
+   */
+int32_t sntp_parse_response( char* response )
+{
+    int32_t current_ntp_time = 0; //epoch + 2208988800
+    int32_t current_epoch_time = 0;
+
+    printf("\r\n>>Parsing SNTP response...");
+    memcpy(&current_ntp_time, response+40, 4); //timestamp starts at byte 40
```

Comment:
This +40 makes me scratch my brain :) Could make a const variable that is named like a timestamp offset please :)